tests/system/robot/symbolPronunciationTests.py

```diff
@@ -353,111 +335,109 @@ def _getCharAfterKey(key, maxWait=0.1) -> str:
-	this function will press the key maxAttempt times until an associated description is found.
-	If no char description is found, it will raise an assertionError.
-	the key should cause an unit character movement for NVDA to read a single character.
-	@param maxAttempts: number of times that the key is pressed and the char is read
-	before raise the exception if an associated description is not found.
+	If no char description is found, it will raise an AssertionError.
+	The key should cause a unit character movement to read a single character.
 	@return: a tuple of the read char, the associated description, and the time when the key was pressed.
 	"""
 	readChars = []
-	for i in range(maxAttempts):
-		s = _timer()
-		spoken = _getCharAfterKey(key.value).lower()
-		readChars.append(spoken)
-		if spoken not in _CHARACTER_DESCRIPTIONS:
-			continue
-		return (spoken, _CHARACTER_DESCRIPTIONS[spoken], s)
+	startTime = _timer()
+	spoken = _getCharAfterKey(key.value, maxWaitSeconds=0.1).lower()
+	readChars.append(spoken)
+	if spoken in _CHARACTER_DESCRIPTIONS:
+		return (spoken, _CHARACTER_DESCRIPTIONS[spoken], startTime)
 	raise AssertionError(
-		f"No associated descriptions were found in {maxAttempts} attempts. Read characters: {readChars}"
+		f"No associated descriptions were found; read characters: {readChars}"
 	)
 
 
-def _pressKeyAndWaitDelaiedDescription(key: Move, maxWait: int):
-	""" Press the specified key, captures the read character, and wait until the delaied description is spoken.
-	the delaied description will be compared with the corresponding entry in a dictionary,
+def _pressKeyAndWaitDelayedDescription(key: Move, maxWaitSeconds: float) -> float:
+	"""
+	Press the specified key, captures the read character, and wait until the delayed description is spoken.
+	The delayed description will be compared with the corresponding entry in a dictionary,
 	if an entry is found for the read character.
 	@return: the time since the key was pressed, until the delayed description was spoken.
-	-1 if the expected delayed description was not spoken.
-	The timer starts after the key is pressed and the char is read,
-	because it seems that the communication via spyLib to NVDA is a little slow.
 	"""
-	_, desc, _ = _pressKeyAndGetCharDescription(key)
-	s = _timer()
-	_builtIn.log(f"waiting for {desc}")
-	if _wait_for_specific_speech(desc, maxWait):
-		return _timer() - s
-	return -1
+	spy = _NvdaLib.getSpyLib()
+	_, characterDescription, startTime = _pressKeyAndGetCharDescription(key)
+	spy.wait_for_specific_speech(
+		characterDescription,
+		maxWaitSeconds=maxWaitSeconds,
+		intervalBetweenSeconds=0.01
+	)
+	return _timer() - startTime
 
 
-def _testDelayedDescription(key: Move, delay: int = 1000, repeatCount: int = 1, threshold: int = 70,) -> bool:
-	""" perform delayed character descriptions tests with with the specified parameters:
+def _testDelayedDescription(key: Move, expectedDelaySeconds: float, thresholdSeconds: float = 1) -> None:
```

Comment:
to simplify these tests:

- I have removed the repeat count, performing a test successfully once should suffice.
- Upped the threshold. Devices will vary in performance. 1s max delay is acceptable (this allows a user to set the delay to 50ms and experience at most a delay of ~1s). As mentioned in other comments, the framework and measuring the delay makes up most of that threshold.

Reply:
Hi @seanbudd, all changes are perfect for me. So, I don't have more things to add on this pr.
Thanks for the suggestions, I like this new way in testing robot, thanks for the improved wait_for_specific_speech function.

Let me know if you want me to do something else in this pr.

Regards.